common/src/main/java/whocraft/tardis_refined/common/block/device/LandingPad.java

```diff
@@ -65,13 +65,17 @@ public InteractionResult use(BlockState blockState, Level level, BlockPos blockP
                     if (serverLevel.isEmptyBlock(blockPos.above())) {
                         var tardisLevel = Platform.getServer().getLevel(dimension);
                         TardisLevelOperator.get(tardisLevel).ifPresent(cap -> {
-                            if (cap.getControlManager().beginFlight(true)) {
-                                cap.getControlManager().setTargetLocation(new TardisNavLocation(blockPos.above(), player.getDirection().getOpposite(), serverLevel));
-                                serverLevel.playSound(null, blockPos, SoundEvents.PLAYER_LEVELUP, SoundSource.BLOCKS, 1f, 1f);
-                            } else {
+
+                            if (cap.getControlManager().isOnCooldown()) {
                                 serverLevel.playSound(null, blockPos, SoundEvents.NOTE_BLOCK_BIT, SoundSource.BLOCKS, 100, (float)(0.1 + (serverLevel.getRandom().nextFloat() * 0.5)));
+                            } else {
```

Comment:
Code is good. But try to avoid chained else if statements instead return early where applicable.